### PR TITLE
Improve and expose SPI devices for Linux.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/spi_linux.cc [hunk 1]
```diff
@@ -32,12 +33,34 @@ extern Object* return_open_error(Process* process, int err);
...
+  ARGS(cstring, pathname, int, frequency, int, mode);
+  if (frequency <= 0) FAIL(INVALID_ARGUMENT);
   // We always set the close-on-exec flag otherwise we leak descriptors when we fork.
   // File descriptors that are intended for subprocesses have the flags cleared.
   int fd = open(pathname, O_CLOEXEC | O_RDWR);
   if (fd < 0) return return_open_error(process, errno);
+  int result = ioctl(fd, SPI_IOC_WR_MAX_SPEED_HZ, &frequency);
+  if (result < 0) {
+    close(fd);
+    return return_open_error(process, errno);
```
Comment (floitsch) [inline src/resources/spi_linux.cc:47]: Not sure about using "return_open_error" here.
Do we have something that gets the perror?

Comment (kasperl) [reply src/resources/spi_linux.cc:47]: `Primitive::os_error`

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/resources/spi_linux.cc (hunk 1 above)
Comment (floitsch) [reply src/resources/spi_linux.cc]: done.

**Pull request merged**